src/spikeinterface/core/core_tools.py

```diff
@@ -684,3 +684,20 @@ def measure_memory_allocation(measure_in_process: bool = True) -> float:
         memory = mem_info.total - mem_info.available
 
     return memory
+
+
+def is_path_remote(path: str | Path) -> bool:
+    """
+    Returns True if the path is a remote path (e.g., s3:// or gcs://).
+
+    Parameters
+    ----------
+    path : str or Path
+        The path to check.
+
+    Returns
+    -------
+    bool
+        Whether the path is a remote path.
+    """
+    return "s3://" in str(path) or "gcs://" in str(path)
```

Comment:
Another alternative is to check for the store attribute of the object (for example it is not `DirectoryStore`) to determine this:

https://zarr.readthedocs.io/en/stable/api/storage.html#zarr.storage.DirectoryStore

Reply:
@h-mayorquin this is before the zarr object is instantiated. In particular, we need to avoid casting to `Path` a remote string because it messes up with the formatting!

Reply:
In the code that you have the only one that is done before is the one for mode but that can be easily fixed.

The casting to path could be conditione on the file existing in general I think.

```python
folder_path = Path(folder_path) if Path(folder_path).exists() else folder_path
```

But I think this work as well. I don't see how one is better than the other.